Foundation/NSDictionary.swift

```diff
@@ -586,7 +586,8 @@ open class NSMutableDictionary : NSDictionary {
         guard type(of: self) === NSDictionary.self || type(of: self) === NSMutableDictionary.self else {
             NSRequiresConcreteImplementation()
         }
-        _storage[(aKey as! NSObject)] = _SwiftValue.store(anObject)
+        let aKey = (aKey as? NSObject) ?? NSString(string: aKey as! String)
```

Comment:
Please do not special-case String/NSString here. Can you use _SwiftValue.store(aKey) instead?

Reply:
I agree, _SwiftValue.store is the way we are trying to work-around not having bridges and all bridged conversions should be written into the same funnel point

Reply:
Not sure I follow you - It is the aKey as! NSObject which is the problem. Tell me what you propose and I can test it for you.